pyrene/src/components/DateTimePicker/RangeDateTimeInput/RangeDateTimeInput.tsx

```diff
@@ -0,0 +1,377 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  convertDateTypeToString,
+  convertTimeTypeToString,
+  convertToDateTypeObject,
+  convertToTimeTypeObject,
+  convertToUTCtime,
+  convertToZoneTime,
+  DateType,
+  dateTypeToStandardEUDateFormat,
+  getDateTypeFromddmmyyyyWithSep,
+  getFutureDate,
+  getTimeTypeFromhhmmWithSep,
+  isValidDate, isValidTime, isValidTimeZone, standardEUDateFormat, standardEUTimeFormat, TimeType,
+} from '../../../utils/DateUtils';
+import DateTimeInput from '../DateTimeInput/DateTimeInput';
+
+import styles from './RangeDateTimeRangeInput.css';
+
+type OnFunction = (value?: number | [number, number] | null) => void;
+
+export interface RangeProps {
+  startDate?: Date,
+  startDateValue?: string,
+  endDateValue?: string,
+  startTimeValue?: string,
+  endTimeValue?: string,
+  labels?: [string, string],
+  dateOnly?: boolean,
+  errorValue?: string,
+  handleOn?: () => void,
+  invalidTimestamp?: boolean,
+  name?: string,
+  onBlur?: () => void,
+  // onChange?: (date: Date | [Date, Date] | null, event: React.SyntheticEvent<any> | undefined, rangePos?: string) => void,
+  onChange?: OnFunction,
+  onFocus?: (value: string) => void,
+  setEndDateValue?: () => void,
+  setStartDate?: (date:Date | undefined) => void,
+  /**
+   * This is a unix timestamp, which is the number of seconds that have elapsed since Unix epoch
+   */
+  timeStamps?: [number, number] | null,
+  timeZone?: string,
+  /**
+   * This is a timestamp that represents the maximum date allowed by the component
+   */
+  maxDateTime?: number,
+  /**
+   * This is a timestamp that represents the minimum date allowed by the component
+   */
+  minDateTime?: number,
+}
+
+const inRange = (timestampToCheck: number, minimumValue: number, maximumValue: number): number => {
+  if (timestampToCheck < minimumValue) {
+    return -1;
+  }
+  if (timestampToCheck > maximumValue) {
+    return 1;
+  }
+  return 0;
+};
+
+const RangeDateTimeRangeInput: React.FC<RangeProps> = ({
+  name = '',
+  labels = ['Von', 'Bis'],
+  onBlur = () => {},
+  onChange = () => {},
+  onFocus = () => {},
+  dateOnly = false,
+  endDateValue,
+  endTimeValue,
+  startDateValue,
+  startTimeValue,
+  timeStamps,
+  timeZone = 'Europe/Zurich',
+  maxDateTime = getFutureDate({ years: 1 }),
+  minDateTime = 0,
+}: RangeProps) => {
+
+  const [iendDate, setEndDate] = useState<Date | undefined>(undefined);
```

Comment:
When you have plenty of state like you have here, it s recommended to use useReducer. I used it here.